src/entity/trade/TradeRecipe.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\entity\trade;
+
+use pocketmine\item\Item;
+use pocketmine\nbt\tag\CompoundTag;
+
+final class TradeRecipe{
+
+	public const TAG_BUY_A = "buyA";
+	public const TAG_BUY_B = "buyB";
+	public const TAG_SELL = "sell";
+	public const TAG_REWARD_EXP = "rewardExp";
+	public const TAG_TIER = "tier";
+	public const TAG_TRADER_EXP = "traderExp";
+	public const TAG_MAX_USES = "maxUses";
+	public const TAG_PRICE_MULTIPLIER_A = "priceMultiplierA";
+	public const TAG_PRICE_MULTIPLIER_B = "priceMultiplierB";
+	public const TAG_USES = "uses";
+
+	public const TAG_NET_ID = "netId";
```

Comment:
Tag constans must be private if they are not needed outside the class, plugins should not mess with NBT anyway.